aws-datastore/src/main/java/com/amplifyframework/datastore/DataStoreErrorHandler.java

```diff
@@ -15,10 +15,40 @@
 
 package com.amplifyframework.datastore;
 
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.core.Amplify;
 import com.amplifyframework.core.Consumer;
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.logging.Logger;
 
 /**
- * Just a ~type-alias for a consumer of DataStoreException.
+ * Interface to enable custom error handling when encountered with an error from the
+ * GraphQL server. Will be invoked while publishing a mutation for any GraphQL error
+ * that is not of ConflictUnhandled type, which will be handled by {@link DataStoreConflictHandler}.
  */
-public interface DataStoreErrorHandler extends Consumer<DataStoreException> {
+public interface DataStoreErrorHandler extends Consumer<DataStoreError<? extends Model>> {
+    /**
+     * Gets a new instance of the {@link DefaultDataStoreErrorHandler}.
+     * @return A new {@link DefaultDataStoreErrorHandler}
+     */
+    @NonNull
+    static DataStoreErrorHandler defaultHandler() {
+        return new DefaultDataStoreErrorHandler();
+    }
+
+    /**
+     * Default instance for handling error response from the GraphQL server. Error message
+     * is logged, but no further action is taken.
+     */
+    final class DefaultDataStoreErrorHandler implements DataStoreErrorHandler {
```

Comment:
I think I had named these originally. And I apologize. I now have come up with a better idea. 😆 

Maybe we should more clearly name these:
```suggestion
    @NonNull
    static LoggingErrorHandler loggingHandler() {
        return new LoggingErrorHandler();
    }

    /**
     * Default instance for handling error response from the GraphQL server. Error message
     * is logged, but no further action is taken.
     */
    final class LoggingErrorHandler implements DataStoreErrorHandler {
```

Reply:
I will rename `DefaultDataStoreErrorHandler` to `LoggingErrorHandler` as you suggest to make the class name make more sense, but I will keep the static method `defaultHandler()` name because we may decide to change the default behavior in the future.